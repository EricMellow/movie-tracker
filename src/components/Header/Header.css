.header {
  height: 100px;
  margin: 0 auto 80px auto;
  width: 80%;
  display: flex;
  justify-content: space-between;
}
.header a {
  color: #ffffff;
  margin-left: 45px;
  font-size: rem;
  text-decoration: none;
}

nav {
  width: auto;
  display: inline-block;
  margin-top: 95px;
  position: relative;
}

.logo {
  height: 50px;
  margin-top: 60px;
}

.logo:hover {
  cursor: pointer;
}

.favoritesError {
  width: auto;
  background: #333;
  padding: 15px 20px 15px 20px;
  display: inline-block;
  border-radius: 5px 5px 0 0;
  position: absolute;
  top: 35px;
  right: 30px;
  color: #ffba00;
  box-shadow: inset 1px 1px 1px #000;
}

@media only screen and (max-width: 800px) {
.header {
  flex-direction: column;
  height: 100px;
  justify-content: center;
  padding-top: 30px;
}

nav {
  width: 100%;
  margin-top: 45px;
  justify-content: center;
  display: flex;
  position: relative;
}

.logo {
  height: 50px;
  width: 270px;
  margin-top: 60px;
  margin: 0 auto;
  }
}

@media only screen and (max-width: 650px) {
  .favoritesError {
    top: 41px;
    font-size: 0.7rem;
    right: 0;
    padding-left: 27px;
  }
}