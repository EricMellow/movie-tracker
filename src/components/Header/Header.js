import React, { Component } from 'react';
import './Header.css';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from "react-redux";
import { toggleRenderRecent, logout, setSelectedMovieId } from "../../actions/index";

export class Header extends Component {

  handleLoginClick = () => {
    if (this.props.userId) {
      this.props.logout();
      this.props.history.push('/');
    }
  }

  handleFavoritesClick = () => {
    const movieId = this.props.favoriteMovies.length ? this.props.favoriteMovies[0].movie_id : null;
    this.props.setFeaturedMovie(movieId);
    this.props.toggleRender(false);
  }

  handleRecentsClick = () => {
    // const movieId = this.props.recentMovies.length ? this.props.recentMovies[0].movie_id : null;
    // this.props.setFeaturedMovie(movieId);
    this.props.toggleRender(true)
  }

  render() {
    const loginLogoutText = this.props.userId ? 'Sign Out' : 'Sign Up/Login';
    const path = this.props.userId ? '/' : '/login';

    return (
      <div className="header">
        <img src={require('../Header/movie-tracker-logo.png')} className="logo" />
        <nav>
          <NavLink
            to={path}
            className="navLink"
            onClick={this.handleLoginClick}
          >{loginLogoutText}</NavLink>
          <NavLink
            to='/favorites'
            className="navLink"
            onClick={this.handleFavoritesClick}
          >Favorites</NavLink>
          <NavLink
            to='/'
            className="navLink"
            onClick={this.handleRecentsClick}
          >Recent Movies</NavLink>
        </nav>
      </div>
    );
  }
};

const mapDispatchToProps = (dispatch) => ({
  toggleRender: (bool) => dispatch(toggleRenderRecent(bool)),
  logout: () => dispatch(logout()),
  setFeaturedMovie: (id) => dispatch(setSelectedMovieId(id))
});

const mapStateToProps = (state) => ({
  userId: state.userId,
  favoriteMovies: state.favoriteMovies,
  selectedMovieId: state.selectedMovieId
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));

