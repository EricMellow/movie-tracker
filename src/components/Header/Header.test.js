import React from 'react';
import ReactDOM from 'react-dom';
import { Header } from '../Header/Header';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history'

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', ()=> {
    const expected = {
      favoritesError: false
    }

    expect(wrapper.state()).toEqual(expected);
  })

  describe('handleLoginLogoutClick', () => {
    let wrapper;
    let mockLogout;
    let history;
    let mockUserId;

    beforeEach(() => {
      mockLogout = jest.fn();
      history = createMemoryHistory('/')
      mockUserId = 3;

      wrapper = shallow(<Header userId={mockUserId} logout={mockLogout} history={history} />);
    });

    it('should call logout if user is signed in', () => {
      const result = wrapper.instance().props.logout;
      wrapper.instance().setRecentFeaturedMovie = jest.fn();
      wrapper.instance().handleLoginLogoutClick();

      expect(result).toHaveBeenCalled();
    });

    it('should call setRecentFeaturedMovie if user is signed in', () => {
      // const result = wrapper.instance().setRecentFeaturedMovie;
      wrapper.instance().setRecentFeaturedMovie = jest.fn();
      wrapper.instance().handleLoginLogoutClick();

      expect(wrapper.instance().setRecentFeaturedMovie).toHaveBeenCalled();
    });

    it('should call setRecentFeaturedMovie if user is signed in', () => {
      wrapper.instance().setRecentFeaturedMovie = jest.fn();
      wrapper.instance().handleLoginLogoutClick();

      expect(wrapper.instance().setRecentFeaturedMovie).toHaveBeenCalled();
    });
  })

  describe('handleFavoritesClick', () => {
    let wrapper;
    let mockFavoriteMovies;
    let mockToggleRender;

    beforeEach(() => {
      mockFavoriteMovies = [{title: 'Princess Bride'}]
      mockToggleRender = jest.fn();

      wrapper = shallow(<Header favoriteMovies={mockFavoriteMovies} toggleRender={mockToggleRender} />);

      wrapper.instance().setFavoriteFeaturedMovie = jest.fn();
    });

    it('should call setFavoriteFeaturedMovie when there are favorite movies', () => {
      wrapper.instance().handleFavoritesClick();

      const result = wrapper.instance().setFavoriteFeaturedMovie

      expect(result).toHaveBeenCalled();
    });

    it('should call toggleRender when there are favorite movies', () => {
      wrapper.instance().handleFavoritesClick();

      const result = mockToggleRender

      expect(result).toHaveBeenCalledWith(false);
    });
  });

});
  