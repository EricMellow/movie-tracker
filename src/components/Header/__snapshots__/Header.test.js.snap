// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
>
          <img
                    className="logo"
                    src="movie-tracker-logo.png"
          />
</Link>,
        <nav>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navLink"
                    onClick={[Function]}
                    to="/login"
          >
                    Sign Up/Login
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navLink"
                    onClick={[Function]}
                    to="/favorites"
          >
                    Favorites
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="navLink"
                    onClick={[Function]}
                    to="/"
          >
                    Recent Movies
          </NavLink>
</nav>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            className="logo"
            src="movie-tracker-logo.png"
/>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
            "src": "movie-tracker-logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="navLink"
              onClick={[Function]}
              to="/login"
>
              Sign Up/Login
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="navLink"
              onClick={[Function]}
              to="/favorites"
>
              Favorites
</NavLink>,
            null,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="navLink"
              onClick={[Function]}
              to="/"
>
              Recent Movies
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Sign Up/Login",
              "className": "navLink",
              "onClick": [Function],
              "to": "/login",
            },
            "ref": null,
            "rendered": "Sign Up/Login",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Favorites",
              "className": "navLink",
              "onClick": [Function],
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Recent Movies",
              "className": "navLink",
              "onClick": [Function],
              "to": "/",
            },
            "ref": null,
            "rendered": "Recent Movies",
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
>
            <img
                        className="logo"
                        src="movie-tracker-logo.png"
            />
</Link>,
          <nav>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navLink"
                        onClick={[Function]}
                        to="/login"
            >
                        Sign Up/Login
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navLink"
                        onClick={[Function]}
                        to="/favorites"
            >
                        Favorites
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navLink"
                        onClick={[Function]}
                        to="/"
            >
                        Recent Movies
            </NavLink>
</nav>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              className="logo"
              src="movie-tracker-logo.png"
/>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
              "src": "movie-tracker-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="navLink"
                onClick={[Function]}
                to="/login"
>
                Sign Up/Login
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="navLink"
                onClick={[Function]}
                to="/favorites"
>
                Favorites
</NavLink>,
              null,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="navLink"
                onClick={[Function]}
                to="/"
>
                Recent Movies
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Sign Up/Login",
                "className": "navLink",
                "onClick": [Function],
                "to": "/login",
              },
              "ref": null,
              "rendered": "Sign Up/Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Favorites",
                "className": "navLink",
                "onClick": [Function],
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Recent Movies",
                "className": "navLink",
                "onClick": [Function],
                "to": "/",
              },
              "ref": null,
              "rendered": "Recent Movies",
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
